### PR TITLE
feat: 'will-move' event for windows. 

### PR DESCRIPTION
**Pull request by sgd2z**
##### Description of Change
Added a "will-move" event to the BrowserWindow, similar to the "will-resize" event. Windows only.

##### Checklist

- [x] `npm test` passes
(https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)

##### Release Notes
Notes: Added "will-move" event to the BrowserWindow.

### CODE REVIEW

File: docs/api/browser-window.md [hunk 1]
```diff
@@ -1,1532 +1,1543 @@
-# BrowserWindow
```
Comment (ckerr) [inline docs/api/browser-window.md]: A 3000 line diff in the docs...? :smile: 

Please fix these side-effects

File: atom/browser/native_window_views_win.cc [hunk 2]
```diff
@@ -207,9 +207,14 @@ bool NativeWindowViews::PreHandleMSG(UINT message,
...
     }
     case WM_MOVING: {
-      if (!movable_)
+      bool prevent_default = false;
+      NotifyWindowWillMove(gfx::Rect(*reinterpret_cast<RECT*>(l_param)),
+                           &prevent_default);
+      if (!movable_ || prevent_default) {
         ::GetWindowRect(GetAcceleratedWidget(),
                         reinterpret_cast<RECT*>(l_param));
+        return true;
```
Comment (ckerr) [inline atom/browser/native_window_views_win.cc]: What does the changed return value do here?

For the most part this patch seems symmetrical with the `WM_SIZING` handler except that it always returns false.

Comment (sgd2z) [reply atom/browser/native_window_views_win.cc]: Without the change in value, frameless windows can still be moved with -webkit-app-region drag

File: docs/api/browser-window.md (hunk 1 above)
Comment (sgd2z) [reply docs/api/browser-window.md]: Ugh, will do, must be whitespace. Thanks!

File: atom/browser/native_window_views_win.cc (hunk 2 above)
Comment (ckerr) [reply atom/browser/native_window_views_win.cc]: Makes sense. Could you add a short comment in the code to this effect?

Comment (sgd2z) [reply atom/browser/native_window_views_win.cc]: @ckerr  done

File: docs/api/browser-window.md (hunk 1 above)
Comment (sgd2z) [reply docs/api/browser-window.md]: @ckerr fixed

**Comment by sgd2z** [comment]
Any word on this? Is there anything else I need to do? Also, what would it take to get "will-resize" and this "will-move" into 3.0.X? We would like to deliver a product that uses stable electron as opposed to something that uses a 4.0 nightly. Thanks!

**Comment by ckerr** [comment]
> Also, what would it take to get "will-resize" and this "will-move" into 3.0.X? We would like to deliver a product that uses stable electron as opposed to something that uses a 4.0 nightly. Thanks!

That would require a change in [Electron's versioning policy](https://electronjs.org/docs/tutorial/electron-versioning#beta-releases-and-bug-fixes). The feature set for `3.0.0` was frozen at 3.0.0-beta.1 and so only bug fixes and security fixes are allowed in the 3-0-x line.

After this feature lands in `master`, I'd expect to see it in either `3.1.0` or `4.0.0`, whichever comes first.

**Pull request merged**